arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2285,77 +2258,29 @@ void RestReplicationHandler::handleCommandHoldReadLockCollection() {
     return;
   }
 
-  {
-    CONDITION_LOCKER(locker, _condVar);
-    _holdReadLockJobs.emplace(
-        id, std::shared_ptr<SingleCollectionTransaction>(nullptr));
-  }
+  // This is an optional parameter, it may not be set (backwards compatible)
+  // If it is not set it will default to a hard-lock, otherwise we do a
+  // potentially faster soft-lock synchronisation with a smaller hard-lock phase.
 
-  AccessMode::Type access = AccessMode::Type::READ;
-  if (EngineSelectorFeature::ENGINE->typeName() == "rocksdb") {
-    // we need to lock in EXCLUSIVE mode here, because simply locking
-    // in READ mode will not stop other writers in RocksDB. In order
-    // to stop other writers, we need to fetch the EXCLUSIVE lock
-    access = AccessMode::Type::EXCLUSIVE;
+  bool doHardLock = VelocyPackHelper::getBooleanValue(body, "doHardLock", true);
```

Comment:
I find `false` as a default for a boolean more intuitive, so I'd rather use `doSoftLockOnly`, or something similar, instead.